tests/unit/RecipientsTest.php

```diff
@@ -0,0 +1,171 @@
+        $this->assertEquals($response->getArrayCopy(), json_decode(self::jsonMock('RecipientMock'), true));
+    }
+
+    /**
+     * @dataProvider recipientProvider
+     */
+    public function testRecipientUpdate($mock)
+    {
+        $requestsContainer = [];
+        $handler = self::buildHandler($requestsContainer, $mock);
+
+        $client = new Client('apiKey', ['handler' => $handler]);
+
+        $response = $client->recipients()->update([
+            "id" => '1',
+            "transfer_interval" => "monthly",
+            "transfer_day" => 5,
+            "transfer_enabled" => true,
+            "automatic_anticipation_enabled" => true,
+            "anticipatable_volume_percentage" => 100,
+            "bank_account_id" => 123
+        ]);
+
+        $this->assertEquals(Endpoint::PUT, self::getRequestMethod($requestsContainer));
+        $this->assertEquals('/1/recipients/1', self::getRequestUri($requestsContainer));
+        $this->assertEquals($response->getArrayCopy(), json_decode(self::jsonMock('RecipientMock'), true));
+    }
+
+    /**
+     * @dataProvider balanceProvider
+     */
+    public function testRecipientBalance($mock)
+    {
+        $requestsContainer = [];
+        $handler = self::buildHandler($requestsContainer, $mock);
+
+        $client = new Client('apiKey', ['handler' => $handler]);
+
+        $response = $client->recipients()->getBalance(["id" => '1' ]);
+
+        $this->assertEquals(Endpoint::GET, self::getRequestMethod($requestsContainer));
+        $this->assertEquals('/1/recipients/1/balance', self::getRequestUri($requestsContainer));
+        $this->assertEquals($response->getArrayCopy(), json_decode(self::jsonMock('BalanceMock'), true));
+    }
+
+    public function testRecipientBalanceOperations()
+    {
+        $mock = new MockHandler([
+            new Response(200, [], self::jsonMock('BalanceOperationListMock'))
+        ]);
+
+        $requestsContainer = [];
+        $handler = self::buildHandler($requestsContainer, $mock);
+
+        $client = new Client('apiKey', ['handler' => $handler]);
+
+        $response = $client->recipients()->getBalanceOperationList(["id" => '1' ]);
+
+        $this->assertEquals(Endpoint::GET, self::getRequestMethod($requestsContainer));
+        $this->assertEquals('/1/recipients/1/balance/operations', self::getRequestUri($requestsContainer));
+        $this->assertEquals($response->getArrayCopy(), json_decode(self::jsonMock('BalanceOperationListMock'), true));
+    }
+
+    public function testRecipientBalanceOperation()
+    {
+        $mock = new MockHandler([
+            new Response(200, [], self::jsonMock('BalanceOperationMock'))
+        ]);
+
+        $requestsContainer = [];
+        $handler = self::buildHandler($requestsContainer, $mock);
+
+        $client = new Client('apiKey', ['handler' => $handler]);
+
+        $response = $client->recipients()->getBalanceOperation(["recipient_id" => '1', "balance_operation_id" => '1' ]);
```

Comment:
It's good practice to use single quotes when there is no need for string interpolation :wink: